src/project/mesa3d_desktop_intel.rs

```diff
@@ -139,6 +139,37 @@ impl Project for Mesa3DDesktopIntel {
     }
 
     fn extend_module(&self, target: &Path, module: SoongModule) -> SoongModule {
+        let mut soc_specific = SoongProp::Bool(false);
+        for lib in [
+            "libgallium_dri.so",
+            "libvulkan_intel.so",
+            "libGLESv1_CM_mesa.so.1.1.0",
+            "libGLESv2_mesa.so.2.0.0",
+            "libEGL_mesa.so.1.0.0",
+            "pps-producer",
+        ] {
+            if target.ends_with(lib) {
+                soc_specific = SoongProp::Bool(true);
+                break;
+            }
+        }
```

Comment:
I'm trying to avoid mutable objects as much as I can. In that case, it's also adding a `soc_specific` line for each module while most of them do not need it.
Here is what I would propose to avoid the `mut` and having `soc_specific` only for module needing it:
```suggestion
        let is_soc_specific = |module: SoongModule| -> SoongModule {
            for lib in [
                "libgallium_dri.so",
                "libvulkan_intel.so",
                "libGLESv1_CM_mesa.so.1.1.0",
                "libGLESv2_mesa.so.2.0.0",
                "libEGL_mesa.so.1.0.0",
                "pps-producer",
            ] {
                if target.ends_with(lib) {
                    return module.add_prop("soc_specific", SoongProp::Bool(true));
                }
            }
            module
        };
        let module = is_soc_specific(module);
```

Reply:
Thanks! That definitely cleans things up nicely!